test/raft/integration/test_uv_tcp_connect_addrinfo.c

```diff
@@ -0,0 +1,231 @@
+#include "../../../src/raft.h"
```

Comment:
I wonder if there is a way to define the test only for glibc without having to duplicate the rest of the file. The other problem is keeping the CI and the make recipes in sync when adding new tests/binaries.

I see that musl does not define a macro to detect it, but maybe there is a way:
* Either define a macro that gets passed when we want to enable the tests, `gcc -D...`.
* Or identify glibc using `__USE_GNU` and use that to guard the tests.

We just have to #ifdef and skip the test early.

Reply:
> The other problem is keeping the CI and the make recipes in sync when adding new tests/binaries.

I'm on this, running subsets of tests seems to be poorly supported by Automake but we can have an environment variable that causes the addrinfo binary to return without running tests

> We just have to #ifdef and skip the test early.

The problem is that any tests we want to run cannot be linked into the same binary as `test_addrinfo.c` because that file overrides `getaddrinfo` and it's impossible to avoid calling it. The only way to do it would be to effectively ifdef out all of `test_addrinfo.c` along with the tests that use it. If we went this route, rather than detecting glibc or musl I think it would be better to detect how libc is being linked since that's what makes the difference; we'd need to do some compiler flag sniffing or add a new option in configure.ac. I could go either way on this

Reply:
I've reverted the commit that introduced the new test binary and switched to controlling this with a `configure.ac` option (that also causes `-all-static` to be passed) and conditional compilation. We can get away without duplicating or complicating the test source files, which is nice.